### DISCUSSION TITLE
局域网hostname解析方案讨论

### DISCUSSION THREAD

**Discussion by xianren78** [pymumu/smartdns#1460] [General]
目前使用方案openwrt上跑smartdns+openclash
dnsmasq 监听53，上游是smartdns 监听5353，smartdns上游是openclash监听 7874
openclash用meta nameserver-policy实现分流。
防火墙里 iptables -t nat -A PREROUTING -p udp --dport 53 -j REDIRECT --to-ports 5353
smartdns自定义设置里添加
dnsmasq-lease-file /tmp/dhcp.leases
server 192.168.0.1 -group lan -exclude-default-group
域名地址里
nameserver /lan/lan
nameserver /.lan/lan
目前这样，可以实现局域网hostname解析，国内网与国外网解析分流，以及翻墙需求
发帖想讨论一下，这种方案效率如何，有没有更优设计方案。

**Comment by PikuZheng** [comment]
我内网服务器固定ip写死。
说实话不太理解动态ip而且需要计算机名解析的场景。
